pkg/ignition/config.go

```diff
@@ -67,6 +76,31 @@ func File(config *Config) (string, error) {
 			return "", err
 		}
 	}
+	if len(config.DnsServers) > 0 {
+		dnsServers := []string{"[Resolve]"}
+		for _, v := range config.DnsServers {
+			dnsEntry := fmt.Sprintf("%s%s", dnsEqualString, v.String())
+			dnsServers = append(dnsServers, dnsEntry)
+		}
+
+		dnsConf := map[string]interface{}{
+			"storage": map[string]interface{}{
+				"files": []interface{}{map[string]interface{}{
+					"path": dnsConfFile,
+					"mode": fileMode,
+					"contents": map[string]interface{}{
+						"inline": strings.Join(dnsServers, "\n"),
+					},
+				}},
+			},
+		}
+
+		// merge dnsConfiguration with ignition content
+		err := mergo.Merge(ignitionBase, dnsConf, mergo.WithAppendSlice)
```

Comment:
Please use `if err := ... ; err != nil` in one line here.

Reply:
Done